src/rules/keyframes.rs

```diff
@@ -265,6 +266,34 @@ impl<'i> ToCss for KeyframesRule<'i> {
   }
 }
 
+/// A percentage of a given timeline range
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(feature = "visitor", derive(Visit))]
+#[cfg_attr(
+  feature = "serde",
+  derive(serde::Serialize, serde::Deserialize),
+  serde(tag = "type", rename_all = "camelCase")
+)]
+#[cfg_attr(feature = "jsonschema", derive(schemars::JsonSchema))]
+#[cfg_attr(feature = "into_owned", derive(static_self::IntoOwned))]
+pub struct TimelineRangePercentage {
+  /// A named timeline range
+  timeline_range_name: TimelineRangeName,
+  /// The percentage progress between the start and end of the rage
+  percentage: Percentage
+}
+
+impl<'i> Parse<'i> for TimelineRangePercentage {
+  fn parse<'t>(input: &mut Parser<'i, 't>) -> Result<Self, ParseError<'i, ParserError<'i>>> {
+    let timeline_range_name = input.try_parse(TimelineRangeName::parse)?;
+    let percentage = input.try_parse(Percentage::parse)?;
```

Comment:
If they're required, you don't need `try_parse`:

```suggestion
    let timeline_range_name = TimelineRangeName::parse(input)?;
    let percentage = Percentage::parse(input)?;
```